src/client/gameplay.cpp

```diff
@@ -183,13 +183,14 @@ void Gameplay::update()
     // Try create some chunk meshes
     auto &chunks = m_clientState.chunks;
     auto &chunkMgr = m_clientState.chunkManager;
-    for (auto itr = chunks.begin(); itr != chunks.end(); itr++) {
+    for (auto itr = chunks.begin(); itr != chunks.end();) {
         const auto &[position, chunk] = *itr;
-        if (m_chunkRenders.find(position) == m_chunkRenders.end()) {
-            if (chunkMgr.hasNeighbours(position)) {
-                m_chunkRenders.emplace(position, makeChunkMesh(*chunk));
-                chunks.erase(itr);
-            }
+        if (m_chunkRenders.find(position) == m_chunkRenders.end() && chunkMgr.hasNeighbours(position)) {
+            m_chunkRenders.emplace(position, makeChunkMesh(*chunk));
+            chunks.erase(itr++);
```

Comment:
https://en.cppreference.com/w/cpp/container/vector/erase

```suggestion
            chunks.erase(itr);
```

Reply:
Isn't `chunks` and `std::unordered_map`?
```C++
struct ClientState {
    std::array<Entity, 512> entities;
    ChunkPositionMap<Chunk *> chunks;
    ChunkManager chunkManager;

    EngineStatus status = EngineStatus::Ok;
};
```
https://en.cppreference.com/w/cpp/container/unordered_map/erase

Reply:
Yes it is, my mistake :)

But the same rules applies to unordered_map

https://en.cppreference.com/w/cpp/container/unordered_map/erase

Reply:
No, it shouldn't it be `itr = chunks.erase(itr);`, like in the example?

Reply:
Yep my mistake again.

I shouldn't use github when I woke up 10 minutes ago haha